### ISSUE TITLE
[Feature]: access tokens for situations like vast.ai

### ISSUE THREAD

**Issue by blazespinnaker** [wandb/wandb#4673]
### Description

There is a serious security issue with using services like vast.ai and wandb.  Kagglers have complained about having their code and models stolen when using similar systems in china.   

Being able to create a temporary least privilege access token (some kind of append only token with very limited read access) would help ameliorate this situation.  

It's not perfect, but for those of us who want access to cheap high end GPUs, it would really help a lot.  It also leverages wandb fault tolerant capability with interruptable instances quite nicely.

### Suggested Solution

The ability to create a temporary least privilege access token that can only append information and not provide read beyond some basic metadata necessary to do what's required.  Github has an example of this in its new fine grained tokens you can look at.

**Comment by ArtsiomWB** [comment]
Hi @blazespinnaker! Sounds like a great feature request. So to confirm you'd like to be able to share your project with the least privilege token with another person. What kind of access/permissions do you think the least privilege token should provide? Read-only? Maybe have an access to making runs too?

**Comment by blazespinnaker** [comment]
@ArtsiomWB No, I don't want to share my project - sort of the opposite, actually.

1. On vast.ai you or I can rent hosted machines with GPUs very cheaply.  Especially 'interruptable' machines which can really leverage wandb fault tolerant capabilities.

2. Unfortunately, the hosts will have access to my code that I upload to these machines.  In fact, on Kaggle, folks have complained about having their code and models stolen when they do this.

3. So, I'd like to be able to securely (as possible) use wanddb to upload my code to these machines and train, but without giving the host access to authentication credentials that could be used to hack my wandb account.

4. I could do this by using an append only (no read) amazon s3 bucket with limited size and push models/checkpoints there, but wandb seems like the right place to do this.

5. Note there is still some risk to having some code stolen, but that can be minimized by other techniques, such as obfuscation - which is orthogonal to wandb.

**Comment by ArtsiomWB** [comment]
Got it, we do have something similar to this, a [service account](https://docs.wandb.ai/guides/technical-faq/general#what-is-a-service-account-and-why-is-it-useful), do you think this would work for you in your use case?

**Comment by blazespinnaker** [comment]
Thanks folks.  I'll try the service account, it sort of looks like what I'm looking for, assuming it is constrained to 'write' only.  My one concern is that it could be used to perform a denial of service attack on my account.
